### PR TITLE
Nested groups and disabled users

### PR DESCRIPTION
**Pull request by lucasmrod**
- Add README.md
- Add group nested search, LdapGroup._get_nested_users() (only TODO is for 'posixGroups' on OpenLDAP)
- Add enabled/disabled account detection (vendor.py)
- Add setup.py/cfg, moved requirements.txt to requirements/  
- pylinted/documented/reformatted code (+ fixed pyflakes checks)
- Code tested on 389-ds, AD and OpenLDAP.

### CODE REVIEW

File: ldap_reader/reader.py [hunk 1]
```diff
@@ -299,37 +363,100 @@ def _pas_ranged_results_wrapper(self, startrange = None):
...
+        '''
+        Recursive function to get the list of all
+        nested users within this LDAP group.
+        It also processes child groups within this group, via recursion.
+        This method does not work for 'posixGroups' on OpenLDAP.
+            - For AD and OpenLDAP Groups, 'member' attribute is used.
+            - For RHDS Groups, 'uniqueMember' attribute is used.
+        Arguments:
+        ldap_id : string, entry's dn.
+        Returns a list of string, each string is the DN of the user.
```
Comment (merickson) [inline ldap_reader/reader.py:412]: Missing documentation on the depth parameter

**Comment by merickson** [comment]
My major comment is the structure of the vendor module. For pulling its settings from config, it works fine, however as we move into an auto-detected world, how do you propose to work that in? This is why I originally envisioned hanging vendor-specific parts off of the `LdapConnection` object, so we had that to probe natively.

**Comment by lucasmrod** [comment]
We could have an `LDAPVendor` class in which the `__init__` function would do the auto-detection.
Then `LDAPConnection.__init__` would do:
`self.ldap_vendor = vendor.LDAPVendor(conn)` 
and then use `self.ldap_vendor` for functions that are vendor specific (like `LDAPVendor.check_enabled/enabled_attrs/etc`)

**Comment by merickson** [comment]
@lucasmrod , I like that idea. Let's keep it in our back pocket for when we get there.

**Pull request merged**